contracts/contract_transactions.md

```diff
@@ -0,0 +1,68 @@
+[back](./contracts.md)
+## Contract Transactions
+
+Contract transactions are of three types:
+- Create
+- Call
+- Disable
+
+A contract is also a normal account with a balance.
+
+
+### Create Contract Transaction
```

Comment:
What is the address of the contract? Is it set by the owner in the create transaction?

Reply:
Can the account of the contract act by itself (e.g., a spend transaction)? If not, we must check if something is a contract on all other transactions.

If the account of a contract can make other transactions, can it for example register itself as an oracle?

Or is it only the owner that can make a spend transaction from the account? Then this is a delegated signing, i.e., someone else signs the transaction for an account. We don't have that currently.

Reply:
For oracles, an account is first created (by the owner), and then it registers itself as an oracle. This might be a good solution for contracts at the first step.

This can later be extended with a dual signing by the contract (to ensure it can operate) and the creator.

This also removes the need for an owner of the contract. The contract itself will sign it's disabling.

Reply:
The address of the contract is created by the create transaction it is a new unique address. (At least that was my thought.)

Reply:
This needs more discussion. We should probably do it offline.

Reply:
Update after design meeting:
A contract and an account is the same thing.
The create contract instruction will create an account with a new pubkey that no one has the private key to.
There will be an attach contract transaction that will attach code to a "normal" account. This is so that a user can e.g. implement his own wallet functionality.
I will update the spec.